### PR TITLE
Connection Manager

### PR DESCRIPTION
**Pull request by coot**

### CODE REVIEW

**Comment by nfrisby** [comment]
As a reviewer, I'm keen to see the connection-manager `pdf` build. Would you ensure that's rendered in this PR's build? A) I'm not sure where to find the rendered pdf and B) it looks like the hydra eval that I'm guessing would make it is failing with `LaTeX Error: File `ntheorem.sty' not found.`

File: ouroboros-consensus/ouroboros-consensus.cabal [hunk 1]
```diff
@@ -133,7 +133,7 @@ library
                        Ouroboros.Consensus.Node
                        Ouroboros.Consensus.Node.DbLock
                        Ouroboros.Consensus.Node.DbMarker
-                       Ouroboros.Consensus.Node.ErrorPolicy
+                       Ouroboros.Consensus.Node.RethrowPolicy
```
Comment (nfrisby) [inline ouroboros-consensus/ouroboros-consensus.cabal:136]: Please re-alphebatize this due to the name change (ie it should move down one).

File: ouroboros-consensus/src/Ouroboros/Consensus/Network/NodeToNode.hs [hunk 2]
```diff
@@ -679,28 +679,41 @@ initiator miniProtocolParameters version Apps {..} =
...
       version
 
--- | A projection from 'NetworkApplication' to a server-side
--- 'OuroborosApplication' for the node-to-node protocols.
+-- | A bi-directional network applicaiton.
 --
--- See 'initiatorNetworkApplication' for rationale for the @_version@ arg.
-responder
+-- Implementation note: network currently doesn't enable protocols conditional
+-- on the protocol version, but it eventually may; this is why @_version@ is
```
Comment (nfrisby) [inline ouroboros-consensus/src/Ouroboros/Consensus/Network/NodeToNode.hs:685]: The new comment says,

> ...; this is why @_version@ is currently unused.

but this function does pass around its `version` parameter.

Maybe the comment was intended to describe the callbacks that are being passed `version` here.

Please clarify.

File: ouroboros-consensus/src/Ouroboros/Consensus/Node.hs [hunk 3]
```diff
@@ -354,28 +353,30 @@ runWith RunNodeArgs{..} LowLevelRunNodeArgs{..} =
...
+             | (version, blockVersion) <- Map.toList llrnNodeToNodeVersions
+             ]
         , daLocalResponderApplication = combineVersions [
               simpleSingletonVersions
                 version
                 llrnVersionDataNTC
                 (NTC.responder version $ ntcApps blockVersion version)
             | (version, blockVersion) <- Map.toList llrnNodeToClientVersions
             ]
-        , daErrorPolicies = consensusErrorPolicy (Proxy @blk)
```
Comment (nfrisby) [inline ouroboros-consensus/src/Ouroboros/Consensus/Node.hs:378]: Style: please align these `=`s.

**Review by nfrisby — COMMENTED** [review]
Should this PR also remove the `Ouroboros.Consensus.Node.ErrorPolicy.hs` file?

I made a few minor comments.

I'd like to be able to read the pdf before I can understand the changes in a couple of the remaining Consensus files. Thanks.

**Comment by coot** [comment]
> Should this PR also remove the Ouroboros.Consensus.Node.ErrorPolicy.hs file?

We will change this, the idea is that we will provide diffusion for both: non-p2p and p2p version and have a `cardano-node` level switch to enable `p2p`.

**Comment by coot** [comment]
> I'd like to be able to read the pdf before I can understand the changes in a couple of the remaining Consensus files. Thanks.

The pdf just describes the connection manager, the consensus does not need to know about it; but you're welcome to read it :).  You might be also interested in reading about [p2p-governor](https://input-output-hk.github.io/ouroboros-network/ouroboros-network/Ouroboros-Network-PeerSelection-Governor.html).

**Comment by coot** [comment]
The connection manager will not be merged from this branch, another PR will be opened later.

**Pull request closed without merging**